tests/test_arrow_dataset.py

```diff
@@ -776,6 +787,83 @@ def test_flatten(self, in_memory):
                     self.assertNotEqual(dset._fingerprint, fingerprint)
                     assert_arrow_metadata_are_synced_with_dataset_features(dset)
 
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            with Dataset.from_dict(
+                {"a": [{"en": "Thank you", "fr": "Merci"}] * 10, "foo": [1] * 10},
+                features=Features({"a": Translation(languages=["en", "fr"]), "foo": Value("int64")}),
+            ) as dset:
+                with self._to(in_memory, tmp_dir, dset) as dset:
+                    fingerprint = dset._fingerprint
+                    dset.flatten_()
+                    self.assertListEqual(sorted(dset.column_names), ["a.en", "a.fr", "foo"])
+                    self.assertListEqual(sorted(dset.features.keys()), ["a.en", "a.fr", "foo"])
+                    self.assertDictEqual(
+                        dset.features,
+                        Features({"a.en": Value("string"), "a.fr": Value("string"), "foo": Value("int64")}),
+                    )
+                    self.assertNotEqual(dset._fingerprint, fingerprint)
+                    assert_arrow_metadata_are_synced_with_dataset_features(dset)
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            with Dataset.from_dict(
+                {"a": [{"en": "the cat", "fr": ["le chat", "la chatte"], "de": "die katze"}] * 10, "foo": [1] * 10},
+                features=Features(
+                    {"a": TranslationVariableLanguages(languages=["en", "fr", "de"]), "foo": Value("int64")}
+                ),
+            ) as dset:
+                with self._to(in_memory, tmp_dir, dset) as dset:
+                    fingerprint = dset._fingerprint
+                    dset.flatten_()
+                    self.assertListEqual(sorted(dset.column_names), ["a.language", "a.translation", "foo"])
+                    self.assertListEqual(sorted(dset.features.keys()), ["a.language", "a.translation", "foo"])
+                    self.assertDictEqual(
+                        dset.features,
+                        Features(
+                            {
+                                "a.language": Sequence(Value("string")),
+                                "a.translation": Sequence(Value("string")),
+                                "foo": Value("int64"),
+                            }
+                        ),
+                    )
+                    self.assertNotEqual(dset._fingerprint, fingerprint)
+                    assert_arrow_metadata_are_synced_with_dataset_features(dset)
+
+    @require_pil
+    def test_flatten_complex_image(self, in_memory):
+        # decoding turned on
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            with Dataset.from_dict(
+                {"a": [np.arange(4 * 4 * 3).reshape(4, 4, 3)] * 10, "foo": [1] * 10},
+                features=Features({"a": Image(), "foo": Value("int64")}),
```

Comment:
Could this example also have nested features ? This way it completely test the `table_flatten` function